Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -56,6 +56,25 @@ Two-factor authentication (2FA) is standard for apps that allow users to access
 
 The secondary authentication can be performed at login or later in the user's session. For example, after logging in to a banking app with a username and PIN, the user is authorized to perform non-sensitive tasks. Once the user attempts to execute a bank transfer, the second factor ("step-up authentication") must be presented.
 
+Dangers of SMS-OTP
+
+ Although one-time passwords (OTP) sent via SMS are a common second factor for 2-factor authentication, this method has its shortcomings. In 2016, NIST suggested that "Due to the risk that SMS messages may be intercepted or redirected, implementers of new systems SHOULD carefully consider alternative authenticators." Below you will find a list of some related threats and suggestions to avoid successful attacks on SMS-OTP.  
+
+Threats:  
+
+-  Wireless Interception: The adversary can incept SMS messages by abusing femtocells and other known vulnerabilities in the SS7 network.  
+- Trojans: Malware downloaded on a device used to forward data sent in SMS messages to another number.  
+- SIM SWAP Attack: In this attack, the adversary calls the phone company and requests that a number be used with different sim card owned by the adversary. If successful, the adversary can see messages, and possibly use any 2-factor authentication that is sent to access accounts associated with that number.  
```

Comment:
Maybe we can rewrite the last sentence into: "If successful, the adversary can see the SMS messages which are send to the phone number. This includes the messages used in the 2-factor authentication."

Reply:
For SIM Swap, I would add malicious insider at telecom center.

>- Trojans: Malware downloaded on a device used to forward data sent in SMS messages to another number.
To
> - Trojans: Installed malicious applications with access to text messages may forward the OTP to another number or backend.

> Wireless Interception: The adversary can incept SMS messages by abusing femtocells and other known vulnerabilities in the SS7 network.
  
To

> Wireless Interception: The adversary can intercept SMS messages by abusing femtocells and other known vulnerabilities in the SS7 network.

Reply:
Another thing; maybe the SS7 is not the only network that might be problematic? (Given that there might be insecurities in 2G/3G and 4G as well for instance)

Reply:
Just commenting to say I have read the comments, agree, and will make the necessary changes.

Reply:
Please I thoroughly needs this ss7 software application to bypass Nigeria bank OTP verification of an account to assure a successful withdraw of cash